### DISCUSSION TITLE
MPAS buildlib failure in E3SM1.2

### DISCUSSION THREAD

**Discussion by mailesasaki** [E3SM-Project/E3SM#5652] [E3SM model help]
Hi, I'm Maile and I'm a graduate student trying to port E3SM v.1.2 to Keeling, which is our local machine at University of Illinois, Urbana-Champaign.

I followed the instructions from the [E3SM documentation](https://acme-climate.atlassian.net/wiki/spaces/DOC/pages/7996902/Installing+E3SM+on+a+cluster+or+HPC+platform), where we have cloned the [E3SM git repository,](https://github.com/E3SM-Project/E3SM.git) and have made edits to the config_machines.xml, config_compilers.xml, and the config_batch.xml settings to match our local machine. 

As I try building a case (with ./case.build), the run terminates as it crashes in compiling the `mpas_io.f90` code, where it returns the following message: 

"""
 mpas_io.f90(33): error #6592: This symbol must be a defined parameter, an enumerator, or an argument of an inquiry function that evaluates to a compile-time constant.   [NF_FILL_INT]

mpas_io.f90(34): error #6592: This symbol must be a defined parameter, an enumerator, or an argument of an inquiry function that evaluates to a compile-time constant.   [NF_FILL_CHAR]

mpas_io.f90(36): error #6592: This symbol must be a defined parameter, an enumerator, or an argument of an inquiry function that evaluates to a compile-time constant.   [NF_FILL_DOUBLE]

mpaso built in 24.270038 seconds
mpas_io.f90(33): error #6592: This symbol must be a defined parameter, an enumerator, or an argument of an inquiry function that evaluates to a compile-time constant.   [NF_FILL_INT]

mpas_io.f90(34): error #6592: This symbol must be a defined parameter, an enumerator, or an argument of an inquiry function that evaluates to a compile-time constant.   [NF_FILL_CHAR]

mpas_io.f90(36): error #6592: This symbol must be a defined parameter, an enumerator, or an argument of an inquiry function that evaluates to a compile-time constant.   [NF_FILL_DOUBLE]

ERROR: BUILD FAIL: mpaso.buildlib failed, cat /data/keeling/a/mailes2/a/E3SM/test1/run/test1/bld/ocn.bldlog.230427-133446
BUILD FAIL: mpascice.buildlib failed, cat /data/keeling/a/mailes2/a/E3SM/test1/run/test1/bld/ice.bldlog.230427-133446 

"""

The build logs are linked here: 
[bldlog_ice.txt](https://github.com/E3SM-Project/E3SM/files/11376497/bldlog_ice.txt)
[bldlog_ocn.txt](https://github.com/E3SM-Project/E3SM/files/11376498/bldlog_ocn.txt)

I found a related forum discussion on WRF&MPAS from NCAR that mentioned a similar error, and wonder if the errors they suggested are the same as the one we ran into:  https://forum.mmm.ucar.edu/threads/pio-check-fail-during-compilation-of-mpas-7-0.8613/

Is this error I got due to the PIO library not being compiled with support for the parallel-netCDF library, and would I need to download the PIO library, or is the code crashing due to some other problems? 

Loaded modules: 
module load intel/intel-15.0.3
module load intel/hdf5-1.10.6-intel-15.0.3
module load intel/netcdf4-4.7.4-intel-15.0.3
module load intel/openmpi-4.1.2-intel-15.0.3

I would be happy to provide any additional details. Thanks!

**Comment by philipwjones** [comment]
@mailesasaki The proximate cause of the error is that the MPAS build is not getting the USE_PIO2 flag so is trying to access the NF_FILL variables instead of the PIO_FILL variables. That may be related to whatever PIO setup you've arranged in your machine configuration, but I'm not as familiar with how that works so don't know where to point you.

BTW, what project are you working on at Illinois?  That has some implications for the level of support we can provide.

**Comment by mailesasaki** [comment]
@philipwjones Hi Philip, thank you for your response! We're on a DOE RGMA project here at UIUC.

**Comment by philipwjones** [comment]
@mailesasaki  Are you required to stay with the v1.2 version or can you upgrade to v2 or v2.1? The pio (scorpio) setup is easier there. But if you need to stick with v1.2, then maybe @jayeshkrishna  can advise on what the most compatible pio/scorpio setup would be for v1.2.

**Reply by mailesasaki** [reply]
Hi Philip, sorry for the late reply! One of my colleagues has been able to port E3SM successfully using v2. Thank you for your guidance!

**Comment by philipwjones** [comment]
Ok, good luck and thanks for letting us know. I'll go ahead and close this.
